common.tf

```diff
@@ -9,16 +9,33 @@ locals {
   apiary_bucket_prefix                 = "${local.instance_alias}-${data.aws_caller_identity.current.account_id}-${data.aws_region.current.name}"
   apiary_assume_role_bucket_prefix     = [for assumerole in var.apiary_assume_roles : "${local.instance_alias}-${data.aws_caller_identity.current.account_id}-${lookup(assumerole, "allow_cross_region_access", false) ? "*" : data.aws_region.current.name}"]
   enable_route53_records               = "${var.apiary_domain_name == "" ? "0" : "1"}"
-  apiary_managed_schema_names_original = [for schema in var.apiary_managed_schemas : "${schema.schema_name}"]
-  apiary_managed_schema_names_replaced = [for schema in var.apiary_managed_schemas : "${replace(schema.schema_name, "_", "-")}"]
-  apiary_data_buckets                  = [for schema in var.apiary_managed_schemas : "${local.apiary_bucket_prefix}-${replace(schema.schema_name, "_", "-")}"]
+  #
+  # Create a new list of maps with some extra attributes needed later
+  #
+  schemas_info = [
+    for schema in var.apiary_managed_schemas: merge(
+      {
+        replaced_name : replace(schema["schema_name"], "_", "-"),
+        data_bucket   : "${local.apiary_bucket_prefix}-${replace(schema["schema_name"], "_", "-")}"
+      },
+      schema)
+  ]
+
+  #
+  # Create some string arrays of commonly-referenced values for things like join()
+  #
+  apiary_managed_schema_names_original = [for schema in local.schemas_info : schema["schema_name"]]
```

Comment:
do we still need these 3 variables? I see you are using schemas_info for some loops

Reply:
Yes, all three are being used in other places still that need to use either `join` on strings, or need to use `count` in their indexing.  I suppose in all places where they are used I could replace the reference to a `local` variable with the actual list comprehension code, but does that seem better to you?

Reply:
Found a way to get rid of all 3 of those derived lists that didn't look clumsy so will remove them.